managers/bosh-manager/DirectorService.js

```diff
@@ -0,0 +1,1112 @@
+
+
+class DirectorService extends BaseDirectorService {
+  constructor(guid, plan) {
+    super(plan);
+    this.guid = guid;
+    this.plan = plan;
+    this.cloudController = cloudController;
+    this.serviceFabrikClient = serviceFabrikClient;
+    this.director = bosh.director;
+    this.networkSegmentIndex = undefined;
+    this.platformManager = undefined;
+    this.backupStore = backupStore;
+    this.agent = new Agent(this.settings.agent);
+    this.prefix = CONST.SERVICE_FABRIK_PREFIX;
+  }
+
+  assignPlatformManager(platformManager) {
+    this.platformManager = platformManager;
+  }
+
+  get platformContext() {
+    return Promise.try(() => this.networkSegmentIndex ? this.deploymentName : this.director.getDeploymentNameForInstanceId(this.guid))
+      .then(deploymentName => this.director.getDeploymentProperty(deploymentName, CONST.PLATFORM_CONTEXT_KEY))
+      .then(context => JSON.parse(context))
+      .catch(NotFound, () => {
+        /* Following is to handle existing deployments. 
+           For them platform-context is not saved in deployment property. Defaults to CF.
+         */
+        logger.warn(`Deployment property '${CONST.PLATFORM_CONTEXT_KEY}' not found for instance '${this.guid}'.\ 
+        Setting default platform as '${CONST.PLATFORM.CF}'`);
+
+        const context = {
+          platform: CONST.PLATFORM.CF
+        };
+        return context;
+      });
+  }
+
+  static get prefix() {
+    return _
+      .reduce(config.directors,
+        (prefix, director) => director.primary === true ? director.prefix : prefix,
+        null) || super.prefix;
+  }
+
+  get deploymentName() {
+    return this.getDirectorDeploymentName(this.guid, this.networkSegmentIndex);
+  }
+
+  getDirectorDeploymentName(guid, networkSegmentIndex) {
+    let subnet = this.subnet ? `_${this.subnet}` : '';
+    return `${this.prefix}${subnet}-${NetworkSegmentIndex.adjust(networkSegmentIndex)}-${guid}`;
+  }
+
+  getNetworkSegmentIndex(deploymentName) {
+    return _.nth(BaseDirectorService.parseDeploymentName(deploymentName, this.subnet), 1);
+  }
+
+  getInstanceGuid(deploymentName) {
+    return _.nth(BaseDirectorService.parseDeploymentName(deploymentName, this.subnet), 2);
+  }
+
+  initialize(operation) {
+    return Promise
+      .try(() => {
+        this.operation = operation.type;
+        if (operation.type === CONST.OPERATION_TYPE.CREATE) {
+          return this.acquireNetworkSegmentIndex(this.guid);
+        }
+        return this.findNetworkSegmentIndex(this.guid);
+      })
+      .tap(networkSegmentIndex => {
+        assert.ok(_.isInteger(networkSegmentIndex), `Network segment index '${networkSegmentIndex}' must be an integer`);
+        this.networkSegmentIndex = networkSegmentIndex;
+      })
+      .tap(() => {
+        if (operation.type === 'delete') {
+          return Promise
+            .all([
+              this.platformManager.preInstanceDeleteOperations({
```

Comment:
this should ideally not be present now with individual managers for each platform.

Reply:
Platform managers would still remain. SF2.0 managers are for the provisioning use case. Platform managers are for the consuming platform specific use case like SG creation. Cant be removed.